odin.qmd

```diff
@@ -0,0 +1,299 @@
+
+```{r}
+sys <- dust2::dust_system_create(sir_ode, pars = list())
+dust2::dust_system_set_state_initial(sys)
+t <- seq(0, 100)
+y <- dust2::dust_system_simulate(sys, t)
+y <- dust2::dust_unpack_state(sys, y)
+
+plot(t, y$I, type = "l", xlab = "Time", ylab = "I")
+```
+
+::: {.notes}
+From the generator object `gen` we can construct a model (`mod`), here using no parameters -- just using default parameters as defined above. We run over a set of times and output the value of the system at a number of intermediate times.
+:::
+
+
+## Stochastic models
+
+:::: {.columns}
+
+::: {.column width="55%"}
+
+```{r}
+#| echo: false
+#| results: "asis"
+r_output(readLines("models/sir-basic.R"))
+```
+
+:::
+
+::: {.column width="45%"}
+\begin{gather*}
+S(t + dt) = S(t) - n_{SI}\\
+I(t + dt) = I(t) + n_{SI} - n_{IR}\\
+R(t) = R(t) + n_{IR}
+\end{gather*}
+
+:::
+
+::::
+
+. . .
+
+* `dt` is a special parameter
+* every variable has `initial` and `update` pair
+
+
+## ...compared with ODE models
+
+:::: {.columns}
+
+::: {.column style="font-size: 75%"}
+```{r}
+#| echo: false
+#| results: "asis"
+r_output(readLines("models/sir-basic.R"))
+```
+:::
+
+::: {.column style="font-size: 60%"}
+```{r}
+#| echo: false
+#| results: "asis"
+r_output(readLines("models/sir-basic-ode.R"))
+```
+
+:::
+
+::::
+
+
+## Compiling with `odin2`
+
+```{r}
+#| echo: false
+#| results: "asis"
+r_output(model_compile_code("sir", "models/sir-basic.R"))
+```
+
+```{r}
+#| include: false
+sir <- odin2::odin("models/sir-basic.R")
+```
+
+
+## Running a single simulation
+
+
+```{r}
+sys <- dust2::dust_system_create(sir, pars = list(), dt = 0.25)
+dust2::dust_system_set_state_initial(sys)
+t <- seq(0, 100)
+y <- dust2::dust_system_simulate(sys, t)
+y <- dust2::dust_unpack_state(sys, y)
+
+plot(t, y$I, type = "l", xlab = "Time", ylab = "Infected population")
+```
+
+
+## Running multiple simulations
+
+```{r}
+sys <- dust2::dust_system_create(sir, pars = list(), n_particles = 50,
+                                 dt = 0.25)
+dust2::dust_system_set_state_initial(sys)
+t <- seq(0, 100)
+y <- dust2::dust_system_simulate(sys, t)
+y <- dust2::dust_unpack_state(sys, y)
+matplot(t, t(y$I), type = "l", lty = 1, col = "#00000044",
+        xlab = "Time", ylab = "Infected population")
+```
+
+
+## Calculating incidence with `zero_every`
+
+```{r}
+#| echo: false
+#| results: "asis"
+r_output(model_compile_code("sir", "models/sir-incidence.R"))
+```
+
+```{r}
+#| include: false
+sir <- odin2::odin("models/sir-incidence.R")
+```
+
+
+## Incidence accumulates then resets
+
+```{r}
+sys <- dust2::dust_system_create(sir, pars = list(), dt = 1 / 128)
+dust2::dust_system_set_state_initial(sys)
+t <- seq(0, 20, by = 1 / 128)
+y <- dust2::dust_system_simulate(sys, t)
+y <- dust2::dust_unpack_state(sys, y)
+
+plot(t[t %% 1 == 0], y$incidence[t %% 1 == 0], type = "l",
+     ylab = "Infection incidence", xlab = "Time")
+lines(t, y$incidence, col = "red")
+```
+
+
+## Using `interpolate`
```

Comment:
Perhaps add a section on "time varying inputs" of which interpolation is one form?